webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -362,8 +363,21 @@ func (v validator) validateImageOnCreate(ctx *pkgctx.WebhookRequestContext, vm *
 func (v validator) validateClass(ctx *pkgctx.WebhookRequestContext, vm *vmopv1.VirtualMachine) field.ErrorList {
 	var allErrs field.ErrorList
 
-	if vm.Spec.ClassName == "" {
-		allErrs = append(allErrs, field.Required(field.NewPath("spec", "className"), ""))
+	f := field.NewPath("spec", "className")
+
+	switch {
+	case vmopv1util.IsClasslessVM(*vm) &&
+		pkgcfg.FromContext(ctx).Features.VMImportNewNet:
+
+		// Restrict creating classless VM resources to privileged users.
+		if !ctx.IsPrivilegedAccount {
```

Comment:
QQ: Is allowing admin users to create these 'sans-name' classes part of the import design?. Do we need it since vmop will synthesize one internally and not really 'create' the resource from the looks of this implementation.

Reply:
It is part of the design as the Mobility service account is a priv user. They will be able to create VM resources with Instance / BIOS UUIDs that match existing VMs. Come to think of it, should we add some logic in the findVm code that insists a VM found by anything other than its inventory path is actually a VM Service VM? Whether that is insisting any VMs we find must have their ManagedBy set to "VM Service" or some other measure, how do we prevent DevOps users from setting `spec.biosUUID` to a value of an existing VM somewhere in vC, outside vCenter? Should we play it safe and make `spec.biosUUID` and the proposed `spec.instanceUUID` privileged fields?

cc @bryanv @aruneshpa

Reply:
I thought `spec.biosUUID` was already going to be a privileged field ?. only that it was recently removed for revisiting -- https://github.com/vmware-tanzu/vm-operator/pull/494.

> how do we prevent DevOps users from setting spec.biosUUID to a value of an existing VM somewhere in vC, outside vCenter? make the proposed `spec.instanceUUID` privileged fields?
Agree with playing it safe. IIUC, the problem here could be that vmop fetches a VM with a devOps specified InstanceUUID and it's not the intended VM to be used.

Reply:
> only that it was recently removed for revisiting -- https://github.com/vmware-tanzu/vm-operator/pull/494.

Yeah, it is still possible to do this. We used to do it for other fields as well. You have to do the validation in the mutation webhook.

Reply:
Resolved conversation and reprotected `spec.biosUUID` with https://github.com/vmware-tanzu/vm-operator/pull/518.